Typing error with Mypy 1.12.0

With Mypy 1.12.0, released 2024-10-14, an error is reported:

```
src/fmu/dataio/dataio.py:92: error: Item "type" of "Union[Any, type]" has no attribute "__args__"  [union-attr]
```

The code is part of the function `_validate_variable`, which will be deprecated next year. It should be enough to mark the line as being ignorable.

--- a/src/fmu/dataio/dataio.py
+++ b/src/fmu/dataio/dataio.py
@@ -89,7 +89,7 @@ def _validate_variable(key: str, value: type, legals: dict[str, str | type]) -> 
     valid_type = eval(legal_key) if isinstance(legal_key, str) else legal_key
 
     try:
-        validcheck = valid_type.__args__
+        validcheck = valid_type.__args__  # type: ignore[union-attr]
     except AttributeError:
         validcheck = valid_type
 

